front-end/playground/src/pages/SearchEquipment/SearchEquipment.js

```diff
@@ -84,16 +87,87 @@ class SearchEquipment extends React.Component {
     }
 
 
-    render() {
+    handleSubmit(event) {
+        event.preventDefault();
+        let info = {}
+
+
+        if(this.state.identifier !== "") {
+            info.nameContains = this.state.identifier
+        }
+
+        if(this.state.sporttype !== "") {
+            info.sport = this.state.sporttype
+        }
+
+
+        if(this.state.useMap) {
+            if(this.state.anchor[0] < this.state.anchor2[0] && this.state.anchor[1] < this.state.anchor2[1]) {
+                info.latitudeBetweenStart = (Math.round(this.state.anchor[0] * 100) / 100).toFixed(4)
+                info.latitudeBetweenEnd = (Math.round(this.state.anchor2[0] * 100) / 100).toFixed(4)
+                info.longitudeBetweenStart = (Math.round(this.state.anchor[1] * 100) / 100).toFixed(4)
+                info.longitudeBetweenEnd = (Math.round(this.state.anchor2[1] * 100) / 100).toFixed(4)
+            } else {
+                alert('The positions of the markers used to draw rectangle are problematic. Check the description again please')
+                return
+            }
+        }
+
+        
+
 
 
-        const resultingEquipments = this.state.equipments.map(event => <EquipmentInfo name={event.name} description={event.description} id={event.event_id} /> )
+        const user = JSON.parse(localStorage.getItem('user')).token;
+
+        axios
+            .post(baseURL, info,  {headers:{'Authorization': `Token ${user}`}}).then((response) => {
+                
+                if(response.status === 200 || response.status === 201) {
+                    //console.log(response)
+                    console.log(response.data.items)
+                    this.setState({ equipments: response.data.items });
+                    this.setState({ hasSearched: true });
+                    /*this.props.history.push({
+                        pathname: '/search-results-page',
+                        state: { events: response.data.items }
+                    })*/
+                    //this.context.events = response.data.items
+                    alert('Search is successful')
+
+                    //window.location.href = '/search-results-page'
+                } else {
+                    alert(response.message)
+                    //alert('Not valid info for an event')
+                }
+            }
+            ).catch((error) => {
+                alert('There is an error. Try again later')
+            })
+
+    }
+
+
+
+    render() {
+
+        
+        
+        const resultingEquipments = this.state.equipments.map(event =>  <EquipmentInfo name={event.name} description={event.description} id={event["@id"]} /> )
```

Comment:
because setstate is an async function, this line may not work properly. there should be a bugfix in the future

Reply:
thanks for the information but this page was originally belong to @38programmer61 I tried to write the code to not break things in his codebase.